pandas/tests/groupby/test_bin_groupby.py

```diff
@@ -51,6 +52,30 @@ def test_series_bin_grouper():
     tm.assert_almost_equal(counts, exp_counts)
 
 
+def assert_block_lengths(x):
+    assert len(x) == len(x._data.blocks[0].mgr_locs)
```

Comment:
If this fails, the assertion errors bubbles up?

Reply:
Yeah

```diff
diff --git a/pandas/tests/groupby/test_bin_groupby.py b/pandas/tests/groupby/test_bin_groupby.py
index 152086c241..b6518c1962 100644
--- a/pandas/tests/groupby/test_bin_groupby.py
+++ b/pandas/tests/groupby/test_bin_groupby.py
@@ -53,7 +53,7 @@ def test_series_bin_grouper():
 
 
 def assert_block_lengths(x):
-    assert len(x) == len(x._data.blocks[0].mgr_locs)
+    assert len(x) == len(x._data.blocks[0].mgr_locs) + 1
     return 0
```

```python
___________________________________________________________________ test_mgr_locs_updated[assert_block_lengths] ____________________________________________________________________

func = <function assert_block_lengths at 0x122354b90>

    @pytest.mark.parametrize("func", [cumsum_max, assert_block_lengths])
    def test_mgr_locs_updated(func):
        # https://github.com/pandas-dev/pandas/issues/31802
        # Some operations may require creating new blocks, which requires
        # valid mgr_locs
        df = pd.DataFrame({"A": ["a", "a", "a"], "B": ["a", "b", "b"], "C": [1, 1, 1]})
>       result = df.groupby(["A", "B"]).agg(func)

pandas/tests/groupby/test_bin_groupby.py:71:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas/core/groupby/generic.py:939: in aggregate
    return self._python_agg_general(func, *args, **kwargs)
pandas/core/groupby/groupby.py:926: in _python_agg_general
    result, counts = self.grouper.agg_series(obj, f)
pandas/core/groupby/ops.py:640: in agg_series
    return self._aggregate_series_fast(obj, func)
pandas/core/groupby/ops.py:665: in _aggregate_series_fast
    result, counts = grouper.get_result()
pandas/_libs/reduction.pyx:377: in pandas._libs.reduction.SeriesGrouper.get_result
    res, initialized = self._apply_to_group(cached_typ, cached_ityp,
pandas/_libs/reduction.pyx:195: in pandas._libs.reduction._BaseGrouper._apply_to_group
    res = self.f(cached_typ)
pandas/core/groupby/groupby.py:913: in <lambda>
    f = lambda x: func(x, *args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

x = Series([], Name: C, dtype: int64)

    def assert_block_lengths(x):
>       assert len(x) == len(x._data.blocks[0].mgr_locs) + 1
E       assert 1 == (1 + 1)
E        +  where 1 = len(0    1\nName: C, dtype: int64)
E        +  and   1 = len(BlockPlacement(slice(0, 1, 1)))
E        +    where BlockPlacement(slice(0, 1, 1)) = IntBlock: 1 dtype: int64.mgr_locs

pandas/tests/groupby/test_bin_groupby.py:56: AssertionError
==================================================================== 1 failed, 1 passed, 9 deselected in 0.24s =====================================================================
```